hera_mc/cm_hookup.py

```diff
@@ -188,28 +186,49 @@ def __add_hookup_timing(self, hookup_dict):
-        parts_col.append(get_part_type[pr_key]['part'].hptype)
-        parts_col.append('Start')
-        parts_col.append('Stop')
-        return parts_col
+    def __get_column_headers(self, huh):
+        """
+        This gets the full set of headers for a future show_hookup that doesn't require the same
+        hookup starting point.  Returns a single column for now (the one/long_column stuff)
+        """
+        hu_col = {}
+        return_one_column_for_now = True
+        if return_one_column_for_now:  # This is to just not find part_dossiers on everything for now
+            lc = [huh.keys()[0], huh[huh.keys()[0]].keys()[0], 0]  # ...and maybe we'll use the full
+            for hk, hu in huh.iteritems():                        # ...dictionary later
+                for pk, pv in hu.iteritems():
+                    if len(pv) > lc[-1]:
+                        lc = [hk, pk, len(pv)]
+            huh = {hk: {pk: huh[lc[0]][lc[1]]}}
+
+        long_column = [huh.keys()[0], huh[huh.keys()[0]].keys()[0], 0]  # Pick first one to start
+        for hk, hu in huh.iteritems():
+            hu_col[hk] = {}
+            for pk, pv in hu.iteritems():
+                hu_col[hk][pk] = []
+                for h in pv:
+                    get_part_type = self.handling.get_part_dossier(hpn=h.upstream_part,
+                                                                   rev=h.up_part_rev,
+                                                                   at_date=self.at_date,
+                                                                   exact_match=True)
+                    pr_key = cm_utils._make_part_key(h.upstream_part, h.up_part_rev)
+                    hu_col[hk][pk].append(get_part_type[pr_key]['part'].hptype)
+                h = pv[-1]
+                get_part_type = self.handling.get_part_dossier(hpn=h.downstream_part,
+                                                               rev=h.down_part_rev,
+                                                               at_date=self.at_date,
+                                                               exact_match=True)
+                pr_key = cm_utils._make_part_key(h.downstream_part, h.down_part_rev)
+                hu_col[hk][pk].append(get_part_type[pr_key]['part'].hptype)
+                hu_col[hk][pk].append('Start')
+                hu_col[hk][pk].append('Stop')
+                # Next two lines for "one column" solution
+                if len(hu_col[hk][pk]) > long_column[-1]:
+                    long_column = [hk, pk, len(hu_col[hk][pk])]
+        return hu_col[long_column[0]][long_column[1]]
 
     def __hookup_add_correlator_levels(self, hookup_dict, testing):
-        warnings.warn("Warning:  correlator levels don't work with new pol hookup scheme yet (CM_HOOKUP[212]).")
+        warnings.warn("Warning:  correlator levels don't work with new pol hookup scheme yet (CM_HOOKUP[210]).")
```

Comment:
It doesn't seem super helpful to include the line numbers in these messages; this is why editors have search features.